fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java

```diff
@@ -949,4 +956,72 @@ public static ConstantOperator inspect_hive_part_info(ConstantOperator name) {
         return ConstantOperator.createVarchar(json);
     }
 
+    /**
+     * Inspect db locks that hold by threads
+     */
+    @ConstantFunction(name = "inspect_lock_owners", argTypes = {}, returnType = VARCHAR, isMetaFunction = true)
+    public static ConstantOperator inspectLockOwners() {
+        Map<String, Database> dbs = GlobalStateMgr.getCurrentState().getFullNameToDb();
+        JsonArray dbLocks = new JsonArray();
+        for (Database db : dbs.values()) {
+            boolean useful = false;
+            JsonObject ownerInfo = new JsonObject();
+            String name = db.getFullName();
+            ownerInfo.addProperty("lockDbName", name);
+
+            QueryableReentrantReadWriteLock lock = db.getLock();
+
+            // holder information
+            if (lock.getOwner() != null) {
+                useful = true;
+                String ownerName = lock.getOwner().getName();
+                long id = lock.getOwner().getId();
+                ownerInfo.addProperty("ownerThreadName", ownerName);
+                ownerInfo.addProperty("ownerThreadId", id);
+                if (lock.isWriteLocked()) {
+                    ownerInfo.addProperty("lockState", "writeLocked");
+                }
+            } else if (lock.getReadLockCount() > 0) {
+                useful = true;
+                ownerInfo.addProperty("lockState", "readLocked");
+                ownerInfo.addProperty("readLockCount", lock.getReadLockCount());
+            }
```

Comment:
Is it better to also type it when owner is null? This way there is more information.